pkg/reconciler/workspace/controller.go

```diff
@@ -289,18 +289,17 @@ func (c *Controller) reconcile(ctx context.Context, workspace *tenancyv1alpha1.W
 		// TODO: something sensible when we have some use for this field
 		workspace.Status.BaseURL = fmt.Sprintf("%s.workspaces.kcp.dev", workspace.Name)
 	}
-	var shouldReSchedule bool
 	if currentShard := workspace.Status.Location.Current; currentShard != "" {
 		// make sure current shard still exists
 		_, err := c.workspaceShardLister.Get(clusters.ToClusterAwareKey(workspace.ClusterName, currentShard))
 		if errors.IsNotFound(err) {
 			klog.Infof("de-scheduling workspace %q from nonexistent shard %q", workspace.Name, currentShard)
-			shouldReSchedule = true
+			workspace.Status.Location.Current = ""
```

Comment:
If for some reason we've gone from 1 shard to 0 shards in the system, will doing this result in the workspace getting patched and us losing the current (i.e. "old") placement? L308-312 would not be exectued because len=0 in this scenario.

Reply:
Yes, and as written this is intended behavior and has a test to that effect.